### PR TITLE
WIP GuideInspector component (pr for code review)

### PR DESCRIPTION
**Pull request by AnjirHossain**
**Description of the problem this pull request fixes**

Introduces new component called GuideInspector, which populates the properties tab with potential guide properties that you can modify such as `title`, `axis`, `labels`, `ticks`, `grid` and others. 

**Code review goals:**
- to determine which guide properties still need to be added to 
  the inspector for modification
- find ways to improve the component's behavior

**This PR includes:**
- [•] Tests
- [•] functional comments
- [•] high level description of any new components

**Steps to functionally test this:**
- Create new group
- Add symbols to group
- Drop pipeline on symbol vectors
- Click on guides from the guides 
  section in sidebar
- Modify in the Properties tab

### CODE REVIEW

File: src/js/actions/inspectorActions.js [hunk 1]
```diff
@@ -46,6 +47,14 @@ function removeLayers(layerIds) {
   };
 }
 
+function selectGuide(guideId, groupId) {
+  return {
+    type: SELECT_GUIDE,
+    id: guideId,
+    groupId: groupId
```
Comment (arvind) [inline src/js/actions/inspectorActions.js]: Why is the `groupId` necessary when selecting a guide?

File: src/js/actions/markActions.js [hunk 2]
```diff
@@ -168,6 +170,13 @@ function bindField(id, field, property) {
   };
 }
 
+function removeAxisFromGroup(axisId, groupId) {
```
Comment (arvind) [inline src/js/actions/markActions.js]: Rather than a separate `REMOVE_AXIS_FROM_GROUP` action, let's have the mark reducer reuse the `DELETE_GUIDE` action from above. This reduces the chance of erroneously calling only `DELETE_GUIDE` without `REMOVE_AXIS_FROM_GROUP`. To be able to do this, we'll want to make sure `DELETE_GUIDE` has all the necessary information for the marks reducer (namely the groupId and guide type (axis/legend)).

File: src/js/components/encodings/GroupSubMenu.jsx [hunk 3]
```diff
@@ -23,11 +27,27 @@ function mapStateToProps(reduxState, ownProps) {
     group: getIn(reduxState, 'marks.' + ownProps.id)
   };
 }
-
 function mapDispatchToProps(dispatch, ownProps) {
   return {
-    select: function(id) {
-      dispatch(selectMark(id));
+    select: function(id, encType) {
+      // either name selectMark selectEncoding or create selectGroup action
```
Comment (arvind) [inline src/js/components/encodings/GroupSubMenu.jsx]: Not sure I understand this comment.

File: src/js/components/encodings/GroupSubMenu.jsx [hunk 4]
```diff
@@ -23,11 +27,27 @@ function mapStateToProps(reduxState, ownProps) {
...
+      switch (encType) {
+        case 'mark':
+          dispatch(selectMark(id));
+          break;
+        case 'guide':
+          dispatch(selectGuide(id, ownProps.id));
```
Comment (arvind) [inline src/js/components/encodings/GroupSubMenu.jsx:37]: If we remove `groupId` from `selectGuide`, make sure to remove the argument here as well.

File: src/js/components/encodings/GroupSubMenu.jsx [hunk 5]
```diff
@@ -69,35 +87,56 @@ var Group = React.createClass({
         iconMarkup = (<Icon glyph={glyph} onClick={click} />);
     return iconMarkup;
   },
-
-  deleteUpdate: function(id) {
+  deleteUpdate: function(id, deletionType) {
```
Comment (arvind) [inline src/js/components/encodings/GroupSubMenu.jsx]: Let's be consistent with variable names. E.g., `deletionType` &rarr; `primType`

File: src/js/components/encodings/LayerList.jsx [hunk 6]
```diff
@@ -25,6 +25,8 @@ function mapStateToProps(reduxState) {
     getClosestGroupId(reduxState, selectedMarkId) :
     sceneId;
 
+  // console.log('guides: ', getIn(reduxState, 'guides').toJS());
+
```
Comment (arvind) [inline src/js/components/encodings/LayerList.jsx]: Delete these two lines.

File: src/js/components/inspectors/Area.jsx [hunk 7]
```diff
@@ -10,6 +10,8 @@ var AreaInspector = React.createClass({
     var props = this.props,
         primitive = props.primitive;
 
+    console.log('prims from area: ', primitive);
+
```
Comment (arvind) [inline src/js/components/inspectors/Area.jsx]: Delete these two lines.

File: src/js/components/inspectors/ExtentProperty.jsx [hunk 8]
```diff
@@ -95,6 +95,7 @@ var ExtentProperty = React.createClass({
       <div>
         <Property name={start}
           type="number"
+          primtype={props.primtype}
```
Comment (arvind) [inline src/js/components/inspectors/ExtentProperty.jsx:98]: To conform with Lyra's coding style, `primtype` should be `primType`.

File: src/js/components/inspectors/Guide.js [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+      Values need for axis:
+      - current orientation (orient)
+      - current properties.axis.stroke.value (color)
+      - current properties.axis.strokeWidth.value (width)
+
+      axis = {
+        orient: primitive
+      }
+    */
```
Comment (arvind) [inline src/js/components/inspectors/Guide.js]: Remove old comments here and above (L36).

File: src/js/components/inspectors/Guide.js [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+      <div>
+        <div className="property-group">
+          <h3>Axis</h3>
+          <Property name="orient"
+            label="Orient"
+            opts={orientOpts}
+            primtype="guides"
+            primitive={primitive}
+            onChange={this.handleChange}
+            type="select" />
```
Comment (arvind) [inline src/js/components/inspectors/Guide.js:49]: To improve readability, perhaps include a blank line between `Property` tags?

File: src/js/components/inspectors/Property.jsx [hunk 11]
```diff
@@ -12,14 +13,24 @@ function mapStateToProps(state, ownProps) {
...
   }
 
-  var markState = getIn(state, 'marks.' + ownProps.primitive._id),
+  var propertyState = getIn(state, ownProps.primtype + '.' + ownProps.primitive._id),
+      updatePropsPath;
+
+  if (ownProps.name) {
+    if (ownProps.primtype === 'marks') {
       updatePropsPath = 'properties.update.' + ownProps.name;
+      console.log('ownProps.name: ', ownProps.name);
```
Comment (arvind) [inline src/js/components/inspectors/Property.jsx]: Delete leftover `console.log` statement.

File: src/js/components/inspectors/Property.jsx [hunk 12]
```diff
@@ -12,14 +13,24 @@ function mapStateToProps(state, ownProps) {
...
   return {
-    field: getIn(markState, updatePropsPath + '.field'),
-    group: getIn(markState, updatePropsPath + '.group'),
-    scale: getIn(markState, updatePropsPath + '.scale'),
-    signal: getIn(markState, updatePropsPath + '.signal')
+    field: getIn(propertyState, updatePropsPath + '.field'),
+    group: getIn(propertyState, updatePropsPath + '.group'),
+    scale: getIn(propertyState, updatePropsPath + '.scale'),
+    signal: getIn(propertyState, updatePropsPath + '.signal'),
+    value: getIn(propertyState, updatePropsPath)  // fill --> value: {signal: ...}
```
Comment (arvind) [inline src/js/components/inspectors/Property.jsx]: Remove stray comment.

File: src/js/components/inspectors/Property.jsx [hunk 13]
```diff
@@ -60,6 +73,9 @@ var Property = React.createClass({
         field = props.field,
         value = state.value,
         disabled = props.disabled || props.group,
+        onChange = props.onChange || this.handleChange,
+        onBlur = props.onBlur,
+        docId = props.id,
```
Comment (arvind) [inline src/js/components/inspectors/Property.jsx:77]: What is the `docId` for? I don't see it used anywhere?

File: src/js/components/inspectors/Rect.jsx [hunk 14]
```diff
@@ -4,33 +4,38 @@ var React = require('react'),
     ExtentProperty = require('./ExtentProperty');
 
 var RectInspector = React.createClass({
+  propTypes: {
+    primitive: React.PropTypes.object
+  },
   render: function() {
     var props = this.props,
-        primitive = props.primitive;
+        primitive = this.props.primitive;
```
Comment (arvind) [inline src/js/components/inspectors/Rect.jsx]: Undo this change, `props` is defined on the previous line.

File: src/js/components/inspectors/Scale.jsx [hunk 15]
```diff
@@ -10,6 +10,8 @@ var ScaleInspector = React.createClass({
   render: function() {
     var scale = this.props.primitive;
 
+    // console.log('primitives from scale: ', scale);
+
```
Comment (arvind) [inline src/js/components/inspectors/Scale.jsx]: Delete these two stray lines.

File: src/js/reducers/guidesReducer.js [hunk 16]
```diff
@@ -16,11 +17,19 @@ function guideReducer(state, action) {
...
+    return deleteKeyFromMap(state, action.id);
+  }
+
   if (action.type === ACTIONS.UPDATE_GUIDE_PROPERTY) {
     return setIn(state, action.id + '.' + action.property,
       Immutable.fromJS(action.value));
   }
 
+  if (action.type === ACTIONS.DELETE_SCALE) {
+    return deleteKeyFromMap(state, action.id); // Immutable Maps
```
Comment (arvind) [inline src/js/reducers/guidesReducer.js]: Delete stray comment.

File: src/js/store/factory/Guide.js [hunk 17]
```diff
@@ -46,7 +98,7 @@ var def = module.exports = function(gtype, type, scale) {
     };
   }
 
-  guide.title = undefined;
+  guide.title = 'title';
```
Comment (arvind) [inline src/js/store/factory/Guide.js]: Revert to `undefined`? Titles get automatically filled in during data binding and, if a user were to create an axis manually, I'd prefer for the title to be blank rather than `title`.

**Comment by arvind** [comment]
Great work, @AnjirHossain! This is a functional +1 and we'll be able to merge it once the comments have been addressed (mostly minor cleanups!).

**Comment by arvind** [comment]
Also, now that I think about it, we should make sure that `primType` is added to all the other mark property inspectors (Symbol, etc.).

File: src/js/components/inspectors/Property.jsx (hunk 13 above)
Comment (AnjirHossain) [inline src/js/components/inspectors/Property.jsx:77]: @YiyangXu and I added this to avoid future merge conflicts, I believe somewhere in her ScaleInspector component she was using an id for dom manipulation; will speak to her again about the need for docId.

**Comment by arvind** [comment]
Merged with the [bugs fixed](https://github.com/vega/lyra/compare/dbcf895de9c31edb602ad19165809afbae56fb4d...e0a8647cd626da4d058cb4e23a693cb7694272ad) that we discussed this morning. Great job, @AnjirHossain!
